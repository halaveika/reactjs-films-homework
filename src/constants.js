export const API_KEY_V3 = '048dcb8174451b5088affcc2b8dc74c2';
export const API_KEY_V4 = 'eyJhbGciOiJIUzI1NiJ9.'
+ 'eyJhdWQiOiIwNDhkY2I4MTc0NDUxYj'
 + 'UwODhhZmZjYzJiOGRjNzRjMiIsInN1YiI6IjYxOTQxMDMxMTI4M2U5MDA0MzU3ZTE0YyIsInNjb3Bl'
  + 'cyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.Bc1hoTPCkmPNDON8nroRzmUTSEbP9UutVxZE2IBbG7A';
export const URL = 'https://api.themoviedb.org/3/';
export const BACKDROP_PATH_URL = 'https://image.tmdb.org/t/p/w500/';
export const URL_YOUTUBE = 'https://www.youtube-nocookie.com/embed/';
export const TRENDING = 'TRENDING';
export const TOPRATED = 'TOPRATED';
export const COMINGSOON = 'COMINGSOON';
export const ADD_GENRES = 'ADD_GENRES';
export const ADD_VIDEO = 'ADD_VIDEO';
export const ADD_DETAILS = 'ADD_DETAILS';
export const SET_LOADING = 'SET_LOADING';
export const ADD_ITEMS = 'ADD_ITEMS';
export const SET_FILTER = 'SET_FILTER';
export const SET_CURRENT_PAGE = 'SET_CURRENT_PAGE';
export const SET_SEARCH_VALUE = 'SET_SEARCH_VALUE';
export const GENRE = 'GENRE';
export const SET_GENRE = 'SET_GENRE';
export const SET_LIST_FLEX_DIRECTION = 'SET_LIST_FLEX_DIRECTION';
export const API_PAGE_SIZE = 20;
export const MAX_SEARCH_RESULT = 10000;
export const MAX_PAGE = 500;
export const NOT_FOUND_IMG_PATH = 'assets/images/not-found.png';
export const NOT_FOUND_BACKDROP_PATH = 'assets/images/cover-image.jpg';
