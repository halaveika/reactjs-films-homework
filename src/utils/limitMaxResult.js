import { MAX_SEARCH_RESULT, MAX_PAGE } from '../constants';

const limitMaxResult = (result) => {
  if (result.total_pages > MAX_PAGE) { result.total_pages = MAX_PAGE; }
  if (result.total_results > MAX_SEARCH_RESULT) { result.total_results = MAX_SEARCH_RESULT; }
  return result;
};

export default limitMaxResult;
