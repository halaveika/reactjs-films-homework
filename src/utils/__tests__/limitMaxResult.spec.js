import limitMaxResult from '../limitMaxResult';
import { MAX_SEARCH_RESULT, MAX_PAGE } from '../../constants';

describe('limitMaxResult', () => {
  test('limitMaxResult should return result', () => {
    const mockResult = {
      total_pages: 20,
      total_results: 400,
    };
    const result = limitMaxResult(mockResult);
    expect(result).toEqual(mockResult);
    expect(result).toMatchSnapshot();
  });

  test('limitMaxResult should return result lower than max', () => {
    const mockResult = {
      total_pages: MAX_PAGE + 1,
      total_results: MAX_SEARCH_RESULT + 1,
    };
    const result = limitMaxResult(mockResult);
    expect(result.total_pages).toBe(MAX_PAGE);
    expect(result.total_results).toBe(MAX_SEARCH_RESULT);
    expect(result).toMatchSnapshot();
  });
});
