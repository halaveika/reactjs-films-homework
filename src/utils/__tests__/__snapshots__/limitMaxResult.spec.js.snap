// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`limitMaxResult limitMaxResult should return result 1`] = `
Object {
  "total_pages": 20,
  "total_results": 400,
}
`;

exports[`limitMaxResult limitMaxResult should return result lower than max 1`] = `
Object {
  "total_pages": 500,
  "total_results": 10000,
}
`;
